### PR TITLE
Adding automation

### PR DESCRIPTION
**Pull request by mmirecki**

### CODE REVIEW

**Comment by mmirecki** [comment]
ci test please

**Comment by booxter** [comment]
This is largely copy pasted code from other repos. How can we get stdci jobs added triggered here before merging the PR?

**Comment by gbenhaim** [comment]
ci test please

**Comment by gbenhaim** [comment]
Jobs were added to Jenkins https://kubevirt-master-stdci-production.apps.ovirt.org/job/kubevirt_bridge-maker_standard-check-pr/

File: automation/test.sh [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+if [[ $TARGET =~ openshift-.* ]]; then
+  export KUBEVIRT_PROVIDER="os-3.11.0-multus"
+else
+  export KUBEVIRT_PROVIDER="k8s-multus-1.12.2"
+fi
+
+export KUBEVIRT_NUM_NODES=2
+
+wait_for_download_lock() {
```
Comment (phoracek) [inline automation/test.sh]: download_lock functions are not used anywhere in the script. Please drop them.

File: automation/test.sh [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+export WORKSPACE="${WORKSPACE:-$PWD}"
+readonly ARTIFACTS_PATH="$WORKSPACE/exported-artifacts"
+
+if [[ $TARGET =~ openshift-.* ]]; then
+  export KUBEVIRT_PROVIDER="os-3.11.0-multus"
+else
+  export KUBEVIRT_PROVIDER="k8s-multus-1.12.2"
+fi
+
+export KUBEVIRT_NUM_NODES=2
```
Comment (phoracek) [inline automation/test.sh]: Unless we need 2 nodes for testing, please drop this (and keep the default of a single node).

File: automation/test.sh [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+make cluster-down
+make cluster-up
+
+# Wait for nodes to become ready
+set +e
+kubectl get nodes --no-headers
+kubectl_rc=$?
+while [ $kubectl_rc -ne 0 ] || [ -n "$(kubectl get nodes --no-headers | grep NotReady)" ]; do
+    echo "Waiting for all nodes to become ready ..."
```
Comment (phoracek) [inline automation/test.sh]: I think we do wait for nodes to sync as a part of `make cluster-up` (i recall seeing this message while bringing the cluster up). Could you please double check?

File: automation/test.sh [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+fi
+
+# Run cluster-sync to deploy bridge-marker on the nodes
+make cluster-sync
+
+timeout=300
+sample=30
+
+for i in ${namespaces[@]}; do
+  # Wait until kubevirt pods are running
```
Comment (phoracek) [inline automation/test.sh]: we don't deploy any kubevirt pods, this block is not needed

File: automation/test.sh (hunk 3 above)
Comment (mmirecki) [reply automation/test.sh]: Output of make cluster-up contains:
+ kubectl --kubeconfig=/etc/kubernetes/admin.conf get nodes --no-headers
node01   NotReady   master   5s    v1.12.2
+ kubectl_rc=0
+ retry_counter=0

Removing

File: automation/test.sh (hunk 2 above)
Comment (mmirecki) [reply automation/test.sh]: Removed

File: automation/test.sh (hunk 1 above)
Comment (mmirecki) [reply automation/test.sh]: Removed

File: automation/test.sh (hunk 4 above)
Comment (mmirecki) [reply automation/test.sh]: Removed

File: automation/check-merged.environment.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+---
```
Comment (SchSeba) [inline automation/check-merged.environment.yaml]: Can we remove this and use travis like all our projects?

This PR https://github.com/kubevirt/bridge-marker/pull/5 should do it already.

Comment (mmirecki) [reply automation/check-merged.environment.yaml]: Removed

**Comment by mmirecki** [comment]
ci test please

File: automation/check-merged.sh [hunk 6]
```diff
@@ -0,0 +1,18 @@
+#!/bin/bash -xe
```
Comment (phoracek) [inline automation/check-merged.sh]: please drop all -merged modules in favor of https://github.com/kubevirt/bridge-marker/pull/5

File: automation/test.sh [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+timeout=300
+sample=30
+
+for i in ${namespaces[@]}; do
+
+  # Make sure all containers are ready
+  current_time=0
+  while [ -n "$(kubectl get pods -n $i -o'custom-columns=status:status.containerStatuses[*].ready' --no-headers | grep false)" ]; do
+    echo "Waiting for KubeVirt containers to become ready ..."
```
Comment (phoracek) [inline automation/test.sh]: we don't deploy any kubevirt compononents, please drop this blocking code.

**Comment by mmirecki** [comment]
ci test please

File: automation/check-merged.sh (hunk 6 above)
Comment (mmirecki) [reply automation/check-merged.sh]: dropped

File: automation/test.sh (hunk 7 above)
Comment (mmirecki) [reply automation/test.sh]: removed

File: automation/test.sh [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+trap '{ make cluster-down; }' EXIT SIGINT SIGTERM SIGSTOP
+
+make cluster-down
+make cluster-up
+
+echo "Nodes are ready:"
+kubectl get nodes
+
+# OpenShift is running important containers under default namespace
+namespaces=(kube-system default)
```
Comment (phoracek) [inline automation/test.sh]: this is not used anywhere.

File: automation/test.sh [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+kubectl() { cluster/kubectl.sh "$@"; }
+
+# Make sure that the VM is properly shut down on exit
+trap '{ make cluster-down; }' EXIT SIGINT SIGTERM SIGSTOP
+
+make cluster-down
+make cluster-up
+
+# Run cluster-sync to deploy bridge-marker on the nodes
+make cluster-sync
```
Comment (phoracek) [inline automation/test.sh]: https://github.com/kubevirt/bridge-marker/blob/master/cluster/sync.sh#L42 is not blocking for the marker to become ready. Please add the waiting logic to that script.

Actually, I've been too harsh, the original waiting code (although described as "for kubevirt", would work for all pods, including the marker). Either bring it back with proper comments, or add nicer waiting code to cluster/sync.sh (wait only for known daemon set).

**Comment by phoracek** [comment]
ci test please

**Comment by phoracek** [comment]
@gbenhaim i set up webhooks and ovirt-infra, but it doesn't work

**Comment by phoracek** [comment]
ci test please

**Comment by phoracek** [comment]
/hold

Until we manage to trigger STD-CI

**Comment by gbenhaim** [comment]
ci test please

**Comment by gbenhaim** [comment]
Had to fix the job name in jenkins.
/hold cancel

**Comment by booxter** [comment]
CI passed. Are we waiting on something here?

**Pull request merged**